base/set.jl

```diff
@@ -289,7 +289,7 @@ function _unique!(f, A::AbstractVector, seen::Set, current::Integer, i::Integer)
         end
         i += 1
     end
-    return resize!(A, current - firstindex(A) + 1)
+    return resize!(A, current - firstindex(A) + 1)::typeof(A)
```

Comment:
I've been wondering if we should separate "call API" and "overload API."  It makes much easier to imposing an invariance like this one.  For example, we can have

```julia
function resize!(a, n)
    __resize!__(a, n)
    return a::typeof(a)
end
```

It asserts the return type and also makes sure that the object `a` is returned.  This is useful for something like `push!` where some implementations forget to return the correct object #34274.